src/stores/GuildMemberRoleStore.js

```diff
@@ -137,7 +148,11 @@ class GuildMemberRoleStore extends Collection {
    *   .then(member => console.log(`Member roles is now of ${member.roles.size} size`))
    *   .catch(console.error);
    */
-  set(roles, reason) {
+  set(roles = [], reason) {
```

Comment:
Why are you defaulting to an empty array? That would make `set()` clear all roles except managed ones, while before it wouldn't do anything. Is this something we want to change?

Reply:
if you do <roleStore>.set() it clears off roles regardless, in order to allow for modification i default an array to be converted into a collection, if it was undefined then .map would return an error.

Reply:
> if you do {roleStore}.set() it clears off roles regardless

Inaccurate. As of current master, doing `member.roles.set()` will do.. nothing, as in GuildMember#edit, data.roles is undefined, thereby doing nothing.

![image](https://user-images.githubusercontent.com/17960496/60158089-f59e8100-97f8-11e9-95bf-70f838dc7457.png)

After your change, `member.roles.set()` would in fact clear all roles except managed ones.

> i default an array to be converted into a collection, if it was undefined then .map would return an error

If I pass in anything other than an array or collection (null, undefined, true, false, an object, you name it) it will throw an error. That is still the case with GuildMember#edit, there's no checking for it, and this call will throw. Is that mostly accurate to the current behavior of GuildMember#edit? Absolutely. But this could use some validation, like in add/remove I would think.

Reply:
It used to be that way, it seems I was mistaken, I stopped using a blank .set() awhile ago it seems (i should have double checked) and only use some form of an array, in this case you are correct, however calling a member.edit() would be quite pointless, i'm not entirely sure why it doesn't simply return Promise.resolve(this.member)